docs/gene_sets.md

```diff
@@ -0,0 +1,103 @@
+
+---
+
+### `gene_set_name`
+
+The `gene_set_name` column MUST contain a value. The values for `gene_set_name` MUST NOT contain illegal ASCII characters or sequences. If the following cases are detected, validation MUST display an error message and fail the upload:
+
+* control characters (decimal 0-31)
+* DEL (decimal 127)
+* leading spaces (decimal 32) in a field - "     This is an example"
+* trailing spaces (decimal 32) in a field - "This is an example     " 
+* multiple spaces (decimal 32) "internal" to a field - "This     is an example"
+
+If the `gene_set_name` is missing, validation will display an error message and fail the upload. This is illustrated by **~~?~~** in the example:
+
+| gene_set_name | gene_set_description | gene_symbol | gene_description |
+|---------------|----------------------|-------------|------------------|
+| club.cell     | description          | CCKAR       | description      |
+|     **~~?~~** |                      | SCGB3A2     | description      |
+| club.cell     |                      | CYP2F2      | description      |
+
+
+`gene_symbol(s)` for a `gene_set_name` MAY exist on noncontiguous rows. An out-of-order `gene_symbol` MUST be added to an existing `gene_set_name`. This is illustrated in the example, where **~~CD163~~** is added to the **club.cell** gene set:
+
+| gene_set_name | gene_set_description | gene_symbol | gene_description |
+|---------------|----------------------|-------------|------------------|
+| club.cell     | description          | CCKAR       | description      |
+| club.cell     |                      | SCGB3A2     | description      |
+| club.cell     |                      | CYP2F2      | description      |
+| macrophage    | description          | CD68        | description      |
+| club.cell     |                      | **~~CD163~~**       | description      |
+
+---
+
+### `gene_set_description`
+
+`gene_set_description` MAY contain a value. The first instance of a `gene_set_description` column for a specific `gene_set_name` will be surfaced when a user hovers over `gene_set_name` in cellxgene. All other instances are ignored in subsequent rows for the same `gene_set_name`.
+
+---
+
+### `gene_symbol`
+
+The `gene_symbol` column MUST contain a value that is unique for the `gene_set_name`. Upload will fail and display an error message for a duplicate `gene_symbol`. This is illustrated by **~~CCKAR~~** in the example:
+
+| gene_set_name | gene_set_description | gene_symbol | gene_description |
+|---------------|----------------------|-------------|------------------|
+| club.cell     | description          | CCKAR       | description      |
+| club.cell     |                      | SCGB3A2     | description      |
+| club.cell     |                      | **~~CCKAR~~**   | description      |
+
+---
+
+### `gene_description`
+
+The `gene_description` column MAY contain a value and will be surfaced when a user hovers on `gene_symbol` in cellxgene.
```

Comment:
this doc can't decide if it is a user guide (UI, etc) or a data format spec.  So it is struggling on both cases :-)

If you need both, I suggest two docs, or at least two completely separate sub-docs.

For example, from a data format viewpoint, this section should just say "gene description is optional", and the UI is non-normative